tools/CustomTasks/BatchMergeXaml.cs

```diff
@@ -0,0 +1,134 @@
+        private string nextBaseFile = null;
+
+        private void ExecuteForTaskItems(ITaskItem[] items, string targetOSVersion)
+        {
+            MergedDictionary mergedDictionary = MergedDictionary.CreateMergedDicionary();
+            List<string> files = new List<string>();
+            if (nextBaseFile != null)
+            {
+                files.Add(nextBaseFile);
+            }
+            if (items != null)
+            {
+                foreach (ITaskItem item in items)
+                {
+                    string file = item.ItemSpec;
+                    if (File.Exists(file))
+                    {
+                        files.Add(file);
+                    }
+                    else
+                    {
+                        Log.LogError("Can't find page file " + file);
+                    }
+                }
+            }
+
+            int apiVersion = StripNamespaces.universalApiContractVersionMapping[targetOSVersion];
+            MergeAndGenerateXaml(mergedDictionary, files, targetOSVersion.ToLower(), apiVersion);
+        }
+
+        private void MergeAndGenerateXaml(MergedDictionary mergedDictionary, List<string> files, string targetOSVersion, int apiVersion)
+        {
+            Log.LogMessage("Merge and generate xaml Files for target os" + targetOSVersion);
+
+            foreach (string file in files)
+            {
+                try
+                {
+                    mergedDictionary.MergeContent(File.ReadAllText(file));
+                }
+                catch (Exception)
+                {
+                    Log.LogError("Exception found when merge file " + file);
+                    throw;
+                }
+            }
+
+            string content = mergedDictionary.ToString();
+
+            string name = targetOSVersion + PostfixForGeneratedFile;
+            string fullPath = Path.Combine(OutputDirectory, name);
+
+            string prefixedName = targetOSVersion + postfixForPrefixedGeneratedFile;
+            string prefixedFullPath = Path.Combine(OutputDirectory, prefixedName);
+
+            string strippedContent = StripNamespaces.StripNamespaceForAPIVersion(content, apiVersion);
+
+            filesWritten.Add(Utils.RewriteFileIfNecessary(prefixedFullPath, content));
+            filesWritten.Add(Utils.RewriteFileIfNecessary(fullPath, strippedContent));
+
+            nextBaseFile = prefixedFullPath;
+        }
+
+        public override bool Execute()
+        {
+            if (string.IsNullOrEmpty(OutputDirectory) || !Directory.Exists(OutputDirectory))
+            {
+                Log.LogError("OutputDirectory is empty or not existing");
+            }
+
+            if (string.IsNullOrEmpty(PostfixForGeneratedFile) || !PostfixForGeneratedFile.EndsWith(".xaml"))
+            {
+                Log.LogError("PostfixForGeneratedFile is empty or extension name is not .xaml");
```

Comment:
Consider just leaving out the ".xaml" from the postfix and adding it in programmatically - it seems unnecessary to require devs to add it if it always needs to be there.  "generic" and "themeresources" is the only part that changes, so having that be what gets passed in would make sense.

Reply:
removed .xaml